### PR TITLE
Add more type hints

### PR DESCRIPTION
**Pull request by outa**
Add some more type hints now that #1602 is merged.

cc: #1608 

### Checklist

- [X] I have read the [contributing doc](https://github.com/jrnl-org/jrnl/blob/develop/CONTRIBUTING.md).
- [X] I have included a link to the relevant issue number.
- [X] I have checked to ensure there aren't other open [pull requests](../pulls)
  for the same issue.
- [ ] I have written new tests for these changes, as needed.

### CODE REVIEW

File: jrnl/Journal.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def _encrypt(self, text: str) -> bytes:
 
         return self.encryption_method.encrypt(text)
 
-    def open(self, filename=None):
+    def open(self, filename: str | None = None) -> "Journal":
```
Comment (wren) [inline jrnl/Journal.py:99]: Can we use `Optional[str]` instead of `str | None`?

File: jrnl/encryption/__init__.py [hunk 2]
```diff
@@ -17,7 +18,7 @@ def __str__(self) -> str:
     JRNLV2 = "Jrnlv2Encryption"
 
 
-def determine_encryption_method(config: str | bool) -> "BaseEncryption":
+def determine_encryption_method(config: str | bool) -> Type["BaseEncryption"]:
```
Comment (micahellison) [inline jrnl/encryption/__init__.py:22]: Nice catch here!

**Review by wren — CHANGES_REQUESTED** [review]
Thanks for this! I think the only change is to use `Optional` instead of `foo | None`.

File: jrnl/Journal.py (hunk 1 above)
Comment (outa) [reply jrnl/Journal.py:99]: They are equivalent right? So we could, but I have typed this consistently as `str | None` in #1614 
Anything wrong with the new syntax?

**Review by wren — APPROVED** [review]
lgtm

Thank you!

**Pull request merged**